### PR TITLE
task/WG-140: add app information to request for analytics-related logging

### PR DESCRIPTION
**Pull request by nathanfranklin**
## Overview: ##

~This PR adds the application information (i.e. "taggit", "hazmapper") to the request header to be used by the backend for analytics-related logging.~

This PR adds the application information (i.e. "taggit", "hazmapper") to requests to be used for or analytics-related logging.  This PR has been updated to disable added headers and use query params on a single route instead. Headers can't be used due to https://tacc-main.atlassian.net/browse/WG-191 . The use of headers is left in code but commented out; to be re-used in WG-192 when we start using Taips V3.

See https://github.com/TACC-Cloud/geoapi/pull/161 for more details.

## Related Jira tickets: ##

* [WG-140](https://tacc-main.atlassian.net/browse/WG-140)
* [WG-191](https://tacc-main.atlassian.net/browse/WG-191)

## Testing Steps: ##
See https://github.com/TACC-Cloud/geoapi/pull/161

### CODE REVIEW

**Review by sophia-massie — APPROVED** [review]
LGTM!

File: src/app/app.interceptors.ts [hunk 1]
```diff
@@ -68,22 +68,60 @@ export class TokenInterceptor implements HttpInterceptor {
...
+
+      /*request = request.clone({
         setHeaders: {
-          'X-Guest-UUID': guestUuid,
+          'X-Geoapi-Application': 'hazmapper',
         },
-      });
+      });*/
+
+      analytics_params = { ...analytics_params, application: 'hazmapper' };
```
Comment (sophia-massie) [inline src/app/app.interceptors.ts:84]: Need to change application here to 'taggit'

**Review by sophia-massie — CHANGES_REQUESTED** [review]

File: src/app/app.interceptors.ts (hunk 1 above)
Comment (nathanfranklin) [reply src/app/app.interceptors.ts:84]: 😬  thanks 👍

**Review by sophia-massie — APPROVED** [review]
LGTM

File: src/app/app.interceptors.ts [hunk 2]
```diff
@@ -68,22 +68,46 @@ export class TokenInterceptor implements HttpInterceptor {
...
-        guestUuid = uuidv4();
-        localStorage.setItem('guestUuid', guestUuid);
+      // Using query params instead of custom headers due to https://tacc-main.atlassian.net/browse/WG-191
+      let analytics_params = {};
+
+      analytics_params = { ...analytics_params, application: 'taggit' };
+
+      // for guest users, add a unique id
+      if (!this.authSvc.isLoggedIn()) {
+        // Get (or create of needed) the guestUserID in local storage
```
Comment (taoteg) [inline src/app/app.interceptors.ts:81]: Same typo as related PR: "...(or create _if_ needed)...". ;)

**Review by taoteg — APPROVED** [review]
LGTM! I did not run the code locally, only reviewed in the PR. One minor typo identified in the comments.

File: src/app/app.interceptors.ts (hunk 2 above)
Comment (nathanfranklin) [reply src/app/app.interceptors.ts]: thanks!  fixed 👍

**Pull request merged**